cmd/src/batch_common.go

```diff
@@ -251,6 +255,15 @@ type executeBatchSpecOpts struct {
 	client api.Client
 }
 
+func createDockerWatchdog(ctx context.Context, execUI ui.ExecUI) *watchdog.WatchDog {
+	return watchdog.New(dockerWatchDuration, func() {
+		_, err := docker.CurrentContext(ctx)
```

Comment:
I'm not sure this actually goes back to the Docker Engine in all cases — in theory, the client can answer this from its own configuration. (I briefly tested this by stopping Docker Desktop and running `docker context ls` on the command line, but I haven't put much more effort in than that, admittedly.)

Should we call `NCPU` to be on the safe side? That one _definitely_ has to go to the server.

Reply:
Ha! Nice catch - I definitely didn't know that.